viewer/packages/tools/src/Toolbar/Toolbar.ts

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import { Cognite3DViewer } from '@reveal/core';
+import css from './Toolbar.css';
+
+/**
+ * Toolbar position within the canvas
+ */
+export enum ToolbarPosition {
+  Top = 'Top',
+  Bottom = 'Bottom',
+  Left = 'Left',
+  Right = 'Right'
+}
+
+/**
+ * Toolbar class which creates a container to hold all icons within the toolbar
+ */
+export class Toolbar {
+  private _toolbarContainer: HTMLDivElement;
+  private static readonly stylesId = 'reveal-viewer-toolbar-styles';
+  private readonly _canvasElement: HTMLCanvasElement | null;
+
+  private static readonly classnames = {
+    container: 'reveal-viewer-toolbar-container',
+    bottom: 'reveal-viewer-toolbar-container--bottom',
+    top: 'reveal-viewer-toolbar-container--top',
+    left: 'reveal-viewer-toolbar-container--left',
+    right: 'reveal-viewer-toolbar-container--right',
+    icon: 'reveal-viewer-toolbar-icon',
+    iconImg: 'reveal-viewer-toolbar-icon-img',
+    iconClicked: 'reveal-viewer-toolbar-icon-clicked'
+  };
+
+  private _activeContainerPosition = Toolbar.classnames.bottom;
+
+  constructor(viewer: Cognite3DViewer) {
+    this._canvasElement = viewer.domElement.querySelector('canvas');
+    const canvasElementParent = this._canvasElement?.parentElement;
+    if (canvasElementParent === null) {
+      throw new Error('Could not find canvas');
+    }
+
+    this._toolbarContainer = document.createElement('div');
+    this.createToolBar(canvasElementParent!);
+  }
+
+  /**
+   * Load the the styles from the CSS and appends them to the toolbar
+   * @returns Return if styles already loaded
+   */
+  private static loadStyles() {
+    if (document.getElementById(Toolbar.stylesId)) {
+      return;
+    }
+
+    const style = document.createElement('style');
+    style.id = Toolbar.stylesId;
+    style.appendChild(document.createTextNode(css));
+    document.head.appendChild(style);
+  }
+
+  /**
+   * Creates Toolbar container and adds it to the Canvas parent element and assigns default styles to them
+   * @param controlDiv Canvas parent div element
+   */
+  private createToolBar(controlDiv: HTMLElement) {
+    this._toolbarContainer.id = 'toolbarContainer';
+    Toolbar.loadStyles();
+    this._toolbarContainer.className = Toolbar.classnames.container;
+    this._toolbarContainer.classList.add(Toolbar.classnames.bottom);
+
+    controlDiv.appendChild(this._toolbarContainer);
+  }
+
+  /**
+   * Create & adds a button icons into the toolbar container
+   * @param toolTip Tooltip message to added for the icon
+   * @param backgroundImage Icon image to be displayed
+   * @param isToggle Is the icon button used as toggle
+   * @param onClick Click event action for the icon
+   */
+  public addToolbarItem(toolTip: string, backgroundImage: string, isToggle: boolean, onClick: () => void): void {
```

Comment:
We are using image as `string` with base64 format not as url

Reply:
```
    const iconImage = new Image();
    iconImage.className = Toolbar.classnames.iconImg;
    iconImage.src = backgroundImage;
```

It's used like this so it is just an URI. It _can_ be a base64 encoded image, or it can be an URI. So,
`backgroundImageURI` is probably the right name (i.e. not URL).

However, I'm wondering if it's correct to enforce this - there might be cases where users would want to provide icons in another way, e.g. a SVG. Not sure how to deal with this, but we should think about it